### PR TITLE
Make ConvertAll.from static to avoid pointless allocations, partially addresses #6799

### PR DESCRIPTION
**Pull request by angusholder**
I've got this as 2 commits currently as this change reveals some apparently unnecessary invocations of the conversion routine, but I can't be sure that they're definitely unnecessary without running the test suite to check their removal wouldn't be a breaking change.

I could also now completely dissolve the ConvertAll class into the outer Convert class, but I've erred on the side of making fewer changes rather than more for now, so I'll leave it up to you

### CODE REVIEW

**Comment by lukaseder** [comment]
Thank you very much for your suggestion and for signing the CLA. 

As a general remark (for next time): you should prefix the issue number with the # symbol, so Github will link the commits from the issue #6799 itself. I will comment on the commits directly.

File: jOOQ/src/main/java/org/jooq/tools/Convert.java [hunk 1]
```diff
@@ -380,7 +380,7 @@ else if (from[0] != null && from[0].getClass() == toComponentType) {
      * @throws DataTypeException - When the conversion is not possible
      */
     public static final <T> T convert(Object from, Class<? extends T> toClass) throws DataTypeException {
-        return ConvertAll.from(ConvertAll.from(from, toClass), toClass);
+        return ConvertAll.from(from, toClass);
```
Comment (lukaseder) [inline jOOQ/src/main/java/org/jooq/tools/Convert.java]: This change has been applied on top of your own change, so it is difficult to assess from the change itself whether it is correct.

I do remember there's a case where this was needed, but I can't track it down to an individual regression test right now

**Comment by lukaseder** [comment]
After a quick review, I don't think that the regression risk / benefit are in favour of merging your suggestion. I will hence reject it. Thanks for your understanding.

**Comment by angusholder** [comment]
I was meaning I could revert the second commit if you thought it risked regressions, the first commit alone makes no semantic changes to the code

**Comment by lukaseder** [comment]
> the first commit alone makes no semantic changes to the code

Are you 100% *sure* :)

From my point of view, this is really a cost/benefit ratio assessment right here. Reviewing takes time. Making a very minor improvement without actually proving the benefit is not worth it, e.g. I will have to run the JMC and JMH benchmarks again, because your change doesn't include *proof* that the fix is actually better.

Without such proof, I right now prefer to simply drop the change as this would not have been a priority at all, otherwise.

**Pull request closed without merging**